docs/customization.rst

```diff
@@ -18,99 +198,186 @@ options are:
         'repo_name': 'Material for Sphinx',
         'google_analytics': ['UA-XXXXX','auto'],
         'html_minify': True,
-        'css_minify': True,
         'globaltoc_depth': 2
     }
 
-The complete list of options with detailed explanations appears in
-``theme.conf``.
+Heroes
+==========
+
+To set the hero's text for an individual page, use the ``:hero:`` directive for the desired page.
+If not specified, then the page will not have a hero section.
 
 Configuration Options
```

Comment:
Would be nice to create a sphinx object type for config options.  But that could be done later.

Reply:
I don't fully understand. Do you mean to have a dict like bashtage did in conf.py? Personally, this implementation makes more sense to me than how sphinx-material does it.

Reply:
Sphinx has the concept of an "object type" --- when you document a python function or class, or an environment variable, you create an "object" to refer to the entity that is being documented.  Sphinx defines some built-in object types but you can define your own as well.

The simplest way to define a new object type is with this:
https://www.sphinx-doc.org/en/master/extdev/appapi.html#sphinx.application.Sphinx.add_object_type

Reply:
Sorry, my comment wasn't about `hero` in particular --- I was just suggesting that instead of documenting the supported theme options as an rst definition list, you could instead document the theme options via a new sphinx object type.

For example sphinx-rtd-theme define a `confval` object type here:

https://github.com/readthedocs/sphinx_rtd_theme/blob/c9b1bde560d8ee31400e4e4f92f2e8d7a42265ce/docs/conf.py#L90

And then uses it here to document each config option:

https://github.com/readthedocs/sphinx_rtd_theme/blob/c9b1bde560d8ee31400e4e4f92f2e8d7a42265ce/docs/configuring.rst

Reply:
👍🏼 you should raise this idea as an issue (if only a reminder)